### ISSUE TITLE
inline functions are being treated as components

### ISSUE THREAD

**Issue by alex-golubtsov** [yannickcr/eslint-plugin-react#3011]
Very similar to #1919. The only difference is that function is passed as a property of `options` object.

Here is the code:
```tsx
<MyComponent
  options={{
    tabBarIcon: ({ color, size }) => (
      <Icon name="dumbbell" size={size} color={color} />
    )
  }}
/>
```

ESLint wants `color` and `size` to be explicitly defined
![image](https://user-images.githubusercontent.com/1982853/123174636-b5724780-d480-11eb-8f06-3cfe51b1e5b7.png)

but they actually are in `tabBarIcon` prop
![image](https://user-images.githubusercontent.com/1982853/123174660-bc995580-d480-11eb-9daa-654b23f995ae.png)

**Comment by ljharb** [comment]
I agree; this isn't a component because it doesn't start with a capital letter.
